### PR TITLE
designed table and form using magento

### PR DESCRIPTION
**Pull request by vibhor1111**

### CODE REVIEW

File: app/code/local/Excellence/User/Block/User.php [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <?php
-class Excellence_Test_Block_Test extends Mage_Core_Block_Template
+class Excellence_User_Block_User extends Mage_Core_Block_Template
 {
     public function getContent()
     {
```
Comment (mageexxcel) [inline app/code/local/Excellence/User/Block/User.php]: what is the purpose of this block?

File: app/code/local/Excellence/User/Model/Mysql4/Address.php [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?php
+class Excellence_User_Model_Mysql4_Address extends Mage_Core_Model_Mysql4_Abstract
+{
+    public function _construct()
+    {   
+        $this->_init('user/address', 'test_id');  // here test_id is the primary of the table test2. And test/test2, is the magento table name as mentioned in the       //config.xml file.
```
Comment (mageexxcel) [inline app/code/local/Excellence/User/Model/Mysql4/Address.php]: dont use test_id. use some proper word

File: app/code/local/Excellence/User/controllers/IndexController.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
+<?php
+class Excellence_User_IndexController extends Mage_Core_Controller_Front_Action
+{
+    public function deleteAction()
+    {     
+       if( $this->getRequest()->getParam('id') > 0 ) {
+         try {
+              $model = Mage::getModel('user/user');
```
Comment (mageexxcel) [inline app/code/local/Excellence/User/controllers/IndexController.php]: mode1 and mode2. use proper variables names

File: app/code/local/Excellence/User/controllers/IndexController.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+              $model2 = Mage::getModel('user/address');
+
+              $model->setId($this->getRequest()->getParam('id'))->delete();
```
Comment (mageexxcel) [inline app/code/local/Excellence/User/controllers/IndexController.php]: i dont think this code is correct.
id cannot be same for both model and mode2

File: app/code/local/Excellence/User/controllers/IndexController.php [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+             $this->_redirect('*/*/');
+            } 
+         catch (Exception $e) {
+            Mage::getSingleton('adminhtml/session')->addError($e->getMessage());
+            $this->_redirect('*/*/');
+         }
+        }
+
+        /*================================Added for deleting table 2==================================*/
+        if( $this->getRequest()->getParam('id2') > 0 ) {
```
Comment (mageexxcel) [inline app/code/local/Excellence/User/controllers/IndexController.php]: only pass user_id from table, not id and id2

File: app/code/local/Excellence/User/controllers/IndexController.php [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+            } 
+         catch (Exception $e) {
+            Mage::getSingleton('adminhtml/session')->addError($e->getMessage());
+            $this->_redirect('*/*/');
+         }
+        }
+     }
+public function saveAction() {
+  $fields = $this->getRequest()->getParams();
+  $model = Mage::getModel('user/user')->getCollection();
```
Comment (mageexxcel) [inline app/code/local/Excellence/User/controllers/IndexController.php]: what is use of this line???

File: app/design/frontend/default/models/template/test/table.phtml [hunk 7]
```diff
@@ -0,0 +1,69 @@
+ <!DOCTYPE html>
```
Comment (mageexxcel) [inline app/design/frontend/default/models/template/test/table.phtml]: this is never done in magento.  we never put html, head etc

File: app/design/frontend/default/models/template/test/test.phtml [hunk 8]
```diff
@@ -1,196 +1,65 @@
-<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
+ <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
```
Comment (mageexxcel) [inline app/design/frontend/default/models/template/test/test.phtml]: this is all wrong. see how phtml files are done

File: app/design/frontend/default/models/template/test/test.phtml [hunk 9]
```diff
@@ -1,196 +1,65 @@
...
   {
-    echo '<tr>';
-    echo '<td>'.$data->getTestId().'</td>';
-    echo '<td>'.$data->getFirstname().'</td>';
-    echo '<td>'.$data->getLastname().'</td>';
-    echo '<td>'.$data->getEmail().'</td>';
-    echo '<td>'.$data->getPassword().'</td>';
-    echo '<td>'.$data->getDob().'</td>';
+    $id = $data->getTestId();
+    echo '<tr class="first last odd" >';
```
Comment (mageexxcel) [inline app/design/frontend/default/models/template/test/test.phtml]: echo is never used in phtml file. see core file

File: app/code/local/Excellence/User/Block/User.php [hunk 10]
```diff
@@ -0,0 +1,6 @@
+<?php
+class Excellence_User_Block_User extends Mage_Core_Block_Template
+{
+
```
Comment (mageexxcel) [inline app/code/local/Excellence/User/Block/User.php]: constructor is must

File: app/design/frontend/default/models/layout/test.xml [hunk 11]
```diff
@@ -1,9 +1,28 @@
 <?xml version="1.0"?>
 <layout version="0.1.0">
-    <test_user_view_history>
-        <reference name="content">
-            <block type="test/test" name="test" template="test/test.phtml" />
-        </reference>
-    </test_user_view_history>
+    <user_index_index>
+      <reference name="content">
+         <block type="user/user" name="test" template="test/test.phtml" />
```
Comment (mageexxcel) [inline app/design/frontend/default/models/layout/test.xml]: dont use name test anymore

File: app/design/frontend/default/models/template/test/test.phtml [hunk 12]
```diff
@@ -1,196 +1,60 @@
...
   {
-    echo '<tr>';
-    echo '<td>'.$data->getTestId().'</td>';
-    echo '<td>'.$data->getFirstname().'</td>';
-    echo '<td>'.$data->getLastname().'</td>';
-    echo '<td>'.$data->getEmail().'</td>';
-    echo '<td>'.$data->getPassword().'</td>';
-    echo '<td>'.$data->getDob().'</td>';
+    $id = $data->getUserId();
+    echo '<tr class="first last odd" >';
```
Comment (mageexxcel) [inline app/design/frontend/default/models/template/test/test.phtml]: echo should never be used in phtml files. see core files

File: app/design/frontend/default/models/template/test/test.phtml [hunk 13]
```diff
@@ -1,196 +1,60 @@
...
    <th>DOB</th>
+   <th>CITY</th>
+   <th>PINCODE</th>
+   <th>CONTACT</th>
    <th>Delete</th>
   </tr>
  <?php
- $model = Mage::getModel('test/model1')->getCollection();
- foreach($model as $data)
+ $user_model = Mage::getModel('user/user')->getCollection();
```
Comment (mageexxcel) [inline app/design/frontend/default/models/template/test/test.phtml]: collection should never be called in phtml directly. always call in block

File: app/design/frontend/default/models/layout/user.xml [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+    
+</user_form_printform>
+
+  <default>
+     	<reference name="head">
+     		<action method="addItem"><type>skin_js</type><name>js/script.js</name><params/></action>
+     	</reference>
+  </default>
+  <default>
+       <reference name="head">
```
Comment (sidexcel) [inline app/design/frontend/default/models/layout/user.xml]: no need of bootstrap. remove it

File: app/design/frontend/default/models/template/user/user.phtml [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+
+           <tr class="first last odd" >
+            <td class="a-center last"  >  <?php echo $data->getUserId() ?>  </td>
+            <td class="a-center last"  >  <?php echo $data->getFirstname() ?> </td>
+            <td class="a-center last"  >  <?php echo $data->getLastname() ?> </td>
+            <td class="a-center last"  >  <?php echo $data->getEmail() ?> </td>
+            <td class="a-center last" >  <?php echo $data->getDob() ?>   </td>
+    <?php
+
+      $address_model = $this->getAddressCollection();
```
Comment (sidexcel) [inline app/design/frontend/default/models/template/user/user.phtml]: code formatting!!!

File: app/design/frontend/default/models/template/user/user.phtml [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+     <?php
+
+              }
+            }
+          }
+    
+    ?>
+
+    <p>
+      <b>WANT TO ADD SOME MORE DATA </b>
```
Comment (sidexcel) [inline app/design/frontend/default/models/template/user/user.phtml]: in magento every where always text is written like this
<?php echo __('WANT TO ADD SOME MORE DATA') ?>
follow same format everywhere. see core files to understand

**Pull request merged**